### DISCUSSION TITLE
When loading the YOLOV8 model, it prompts insufficient disk space. What is the reason?

### DISCUSSION THREAD

**Discussion by yyz-git** [beeware/briefcase#2037] [General]
I/python.stdout: Downloading https://github.com/ultralytics/assets/releases/download/v8.3.0/yolov8s-pose.pt to 'yolov8s-pose.pt'...
I/xample.pose_app: Compiler allocated 5703KB to compile void android.view.ViewRootImpl.performTraversals()
W/python.stderr: Error in async handler: WARNING ⚠️ Insufficient free disk space 0.0 GB < 0.000 GB required, Please free 0.0 GB additional disk space and try again.
W/python.stderr: Traceback (most recent call last):
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/toga/handlers.py", line 95, in handler_with_cleanup
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/app/pose_app/app.py", line 37, in take_photo
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/ultralytics/models/yolo/model.py", line 23, in __init__
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/ultralytics/engine/model.py", line 145, in __init__
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/ultralytics/engine/model.py", line 285, in _load
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/ultralytics/nn/tasks.py", line 910, in attempt_load_one_weight
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/ultralytics/nn/tasks.py", line 815, in torch_safe_load
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/ultralytics/utils/downloads.py", line 454, in attempt_download_asset
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/ultralytics/utils/downloads.py", line 331, in safe_download
W/python.stderr:   File "/data/data/com.example.pose_app/files/chaquopy/AssetFinder/requirements/ultralytics/utils/downloads.py", line 231, in check_disk_space
W/python.stderr: MemoryError: WARNING ⚠️ Insufficient free disk space 0.0 GB < 0.000 GB required, Please free 0.0 GB additional disk space and try again.

**Comment by freakboy3742** [comment]
I have no idea what the "YOLOV8" model is  - but the reason for the error about having insufficient disk space is almost certainly ... having insufficient disk space.

The fact that the stack trace references "torch" suggests it might be an AI model - in which case, it's very likely that you're exceeding the 512MB size limit for Android app bundles.

**Reply by yyz-git** [reply]
I am using the YOLOV8S-POSE.PT pose model, which is not large and there is space on the phone's disk. I wonder if it is a storage permission issue? If so, how should I set it up?

**Reply by freakboy3742** [reply]
Again - I have no idea what YOLOV8 *is*, so I can't really provide any detailed advice here. 

If it's an AI model, and it's downloading the model on first use (which is a common approach for AI models, and would be consistent with the stack trace you've provided), then the issue is likely *where* it is trying to save the downloaded model. If the app is trying to download into a location where it doesn't have write permission, then that might manifest as "insufficient disk space".

However - once again - *I know nothing about YOLOv8* - so I can't provide any specific advice about how to configure it to avoid this problem. I'd guess you'd need to configure either the target location where it is downloading, or investigate how to pre-download the model and bundle it with your app so it doesn't need to download at runtime. This is *entirely* in the context of *your* app, and not something that BeeWare is in a position to control.

**Reply by mhsmith** [reply]
Since it apparently thinks you have no free disk space at all, it could also be that it's looking at the correct location, but it isn't able to determine the free disk space on Android. 

If you look at ultralytics/utils/downloads.py, you may be able to come up with a workaround. You can find it somewhere under your app's `build` directory.
